tests/test_port_dpb_system.py

```diff
@@ -248,25 +242,380 @@ def test_port_breakout_with_acl(self, dvs):
+        self.dvs_acl.verify_acl_group_num(0)
 
         # Verify child ports are created.
         self.verify_only_ports_exist(dvs, ["Ethernet0", "Ethernet1", "Ethernet2", "Ethernet3"])
 
-        # Enable below snippet after fixing the above issues
-        '''
         # Move back to 1x100G[40G] mode and verify current mode
         dvs.change_port_breakout_mode("Ethernet0", "1x100G[40G]", "-f")
         dpb.verify_port_breakout_mode(dvs, "Ethernet0", "1x100G[40G]")
-        '''
 
         # Remove ACL table and verify the same
         self.dvs_acl.remove_acl_table("test")
         self.dvs_acl.verify_acl_table_count(0)
 
+    def test_cli_command_with_hyphen_f_option(self, dvs):
+
+        dvs.setup_db()
+        dpb = DPB()
+        self.setup_db(dvs);
+
+        portGroup = ["Ethernet0", "Ethernet1", "Ethernet2", "Ethernet3"]
+        rootPortName = portGroup[0]
+        vlanID = "100"
+        aclTableName = "DPB_ACL_TBL_1"
+        breakoutMode1x = "1x100G[40G]"
+        breakoutMode2x = "2x50G"
+        breakoutMode4x = "4x25G[10G]"
+        breakoutOption = "-f" #Force breakout by deleting dependencies
+
+        # Breakout port with no dependency using "-f" option
+        dvs.change_port_breakout_mode(rootPortName, breakoutMode4x, breakoutOption)
+        dpb.verify_port_breakout_mode(dvs, rootPortName, breakoutMode4x)
+        dvs.change_port_breakout_mode(rootPortName, breakoutMode1x, breakoutOption)
+        dpb.verify_port_breakout_mode(dvs, rootPortName, breakoutMode1x)
+
+        # Breakout port with VLAN and ACL dependency
+
+        # Create ACL table and bind port
+        self.dvs_acl.verify_acl_group_num(0)
+        bind_ports = []
+        bind_ports.append(rootPortName)
+        self.dvs_acl.create_acl_table(aclTableName, "L3", bind_ports)
+        self.dvs_acl.verify_acl_group_num(1)
+
+        # Create VLAN and add port to VLAN
+        self.dvs_vlan.create_vlan(vlanID)
+        self.dvs_vlan.create_vlan_member(vlanID, rootPortName)
+        self.dvs_vlan.get_and_verify_vlan_member_ids(1)
+
+        # Breakout port and make sure it succeeds and associations are removed
+        dpb.verify_port_breakout_mode(dvs, rootPortName, breakoutMode1x)
+        dvs.change_port_breakout_mode(rootPortName, breakoutMode4x, breakoutOption)
+        dpb.verify_port_breakout_mode(dvs, rootPortName, breakoutMode4x)
+        self.dvs_acl.verify_acl_group_num(0)
+        self.dvs_vlan.get_and_verify_vlan_member_ids(0)
+
+        # Add ports to ACL table and VLAN
+        self.dvs_acl.update_acl_table(aclTableName, portGroup)
+        for p in portGroup:
+            self.dvs_vlan.create_vlan_member(vlanID, p)
+        self.dvs_acl.verify_acl_group_num(len(portGroup))
+        self.dvs_vlan.get_and_verify_vlan_member_ids(len(portGroup))
+
+        # Breakout with "-f" option and ensure it succeeds and associations are removed
+        dvs.change_port_breakout_mode(rootPortName, breakoutMode1x, breakoutOption)
```

Comment:
I think, change_port_breakout_mode() should be part of dpb too.

Reply:
I agree. In-fact I am going to move port_dpb.py to dvslib directory and name it as dvs_dpb.py. All DPB related function including breakout, breakin, etc, will be part of that file just like dvs_acl.py, dvs_vlan.py, etc..

I will take it as a sepearate cleanup, until then I will keep it in conftest.py